File: client/CMA/pages/TestingInstitutionManagement/TestingInstitutionInformation/TestingInstitutionInformation.js
// pages/TestingInstitutionManagement/TestingInstitutionInformation/TestingInstitutionInformation.js
//获取全局app实例
const app = getApp()
Page({

  /**
   * 页面的初始数据
   */
  //测试用数据
  data: {
    "testingInstitutionName": "CMA",
    "testingInstitutionAddress": "CMA",
    "postcode": "CMA",
    "fax": "CMA",
    "email": "CMA",
    "tiPeopleInCharge": "CMA",
    "tiPicPosition": "CMA",
    "tiPicTelephone": "CMA",
    "tiPicMobilephone": "CMA",
    "liaison": "CMA",
    "liaisonPosition": "CMA",
    "liaisonTelephone": "CMA",
    "liaisonMobilephone": "CMA",
    "socialCreditCode": "CMA",
    "legalEntityBelongedName": "CMA",
    "legalEntityBelongedAddress": "CMA",
    "lebPeopleInCharge": "CMA",
    "lebPicPosition": "CMA",
    "lebPicTelephone": "CMA",
    "lebSocialCreditCode": "CMA",
    "competentDepartmentName": "CMA",
    "competentDepartmentAddress": "CMA",
    "cdPeopleInCharge": "CMA",
    "cdPicPosition": "CMA",
    "cdPicTelephone": "CMA",
    "characteristic": "CMA",
    "legalEntity": "CMA"
  },
  //修改按钮的处理函数
  viewDetail: function (e) {
    console.log("display -> view")
    //保存指针
    var that = this
    //配置参数并跳转至修改界面
    wx.redirectTo({
      url: '/pages/TestingInstitutionManagement/TestingInstitutionInformation/TestingInstitutionInformationModify/TestingInstitutionInformationModify?testingInstitutionName=' + that.data.testingInstitutionName + "&testingInstitutionAddress=" + that.data.testingInstitutionAddress + "&postcode=" + that.data.postcode + "&fax=" + that.data.fax + "&email=" + that.data.email + "&tiPeopleInCharge=" + that.data.tiPeopleInCharge + "&tiPicPosition=" + that.data.tiPicPosition + "&tiPicTelephone=" + that.data.tiPicTelephone + "&tiPicMobilephone=" + that.data.tiPicMobilephone + "&liaison=" + that.data.liaison + "&liaisonPosition=" + that.data.liaisonPosition + "&liaisonTelephone=" + that.data.liaisonTelephone + "&liaisonMobilephone=" + that.data.liaisonMobilephone + "&socialCreditCode=" + that.data.socialCreditCode + "&legalEntityBelongedName=" + that.data.legalEntityBelongedName + "&legalEntityBelongedAddress=" + that.data.legalEntityBelongedAddress + "&lebPeopleInCharge=" + that.data.lebPeopleInCharge + "&lebPicPosition=" + that.data.lebPicPosition + "&lebPicTelephone=" + that.data.lebPicTelephone + "&lebSocialCreditCode=" + that.data.lebSocialCreditCode + "&competentDepartmentName=" + that.data.competentDepartmentName + "&competentDepartmentAddress=" + that.data.competentDepartmentAddress + "&cdPeopleInCharge=" + that.data.cdPeopleInCharge + "&cdPicPosition=" + that.data.cdPicPosition + "&cdPicTelephone=" + that.data.cdPicTelephone + "&characteristic=" + that.data.characteristic + "&legalEntity=" + that.data.legalEntity,
    })
  },

  /**
   * 生命周期函数--监听页面加载
   */
  onLoad: function (options) {
    //保存指针
    var that = this
    //构造url
    var myurl = app.globalData.url + 'TestingInstitutionInformation/get';
    //请求后端
    app.wxRequest(myurl, 'GET', null, (res) => {
      //成功处理函数
      console.log(res)
      //把接收到的数据存储到页面
      that.setData({
        "testingInstitutionName": res.data.testingInstitutionName,
        "testingInstitutionAddress": res.data.testingInstitutionAddress,
        "postcode": res.data.postcode,
        "fax": res.data.fax,
        "email": res.data.email,
        "tiPeopleInCharge": res.data.tiPeopleInCharge,
        "tiPicPosition": res.data.tiPicPosition,
        "tiPicTelephone": res.data.tiPicTelephone,
        "tiPicMobilephone": res.data.tiPicMobilephone,
        "liaison": res.data.liaison,
        "liaisonPosition": res.data.liaisonPosition,
        "liaisonTelephone": res.data.liaisonTelephone,
        "liaisonMobilephone": res.data.liaisonMobilephone,
        "socialCreditCode": res.data.socialCreditCode,
        "legalEntityBelongedName": res.data.legalEntityBelongedName,
        "legalEntityBelongedAddress": res.data.legalEntityBelongedAddress,
        "lebPeopleInCharge": res.data.lebPeopleInCharge,
        "lebPicPosition": res.data.lebPicPosition,
        "lebPicTelephone": res.data.lebPicTelephone,
        "lebSocialCreditCode": res.data.lebSocialCreditCode,
        "competentDepartmentName": res.data.competentDepartmentName,
        "competentDepartmentAddress": res.data.competentDepartmentAddress,
        "cdPeopleInCharge": res.data.cdPeopleInCharge,
        "cdPicPosition": res.data.cdPicPosition,
        "cdPicTelephone": res.data.cdPicTelephone,
        "characteristic": res.data.characteristic,
        "legalEntity": res.data.legalEntity
      })
      console.log(that.data)
    }, (err) => {
      //失败处理函数
      console.log(err)
    })
  },

  /**
   * 生命周期函数--监听页面初次渲染完成
   */
  onReady: function () {

  },

  /**
   * 生命周期函数--监听页面显示
   */
  onShow: function () {
    //保存指针
    var that = this
    //构造url
    var myurl = app.globalData.url + 'TestingInstitutionInformation/get';
    //请求后端
    app.wxRequest(myurl, 'GET', null, (res) => {
      //成功处理函数
      console.log(res)
      //把接收到的数据存储到页面
      that.setData({
        "testingInstitutionName": res.data.testingInstitutionName,
        "testingInstitutionAddress": res.data.testingInstitutionAddress,
        "postcode": res.data.postcode,
        "fax": res.data.fax,
        "email": res.data.email,
        "tiPeopleInCharge": res.data.tiPeopleInCharge,
        "tiPicPosition": res.data.tiPicPosition,
        "tiPicTelephone": res.data.tiPicTelephone,
        "tiPicMobilephone": res.data.tiPicMobilephone,
        "liaison": res.data.liaison,
        "liaisonPosition": res.data.liaisonPosition,
        "liaisonTelephone": res.data.liaisonTelephone,
        "liaisonMobilephone": res.data.liaisonMobilephone,
        "socialCreditCode": res.data.socialCreditCode,
        "legalEntityBelongedName": res.data.legalEntityBelongedName,
        "legalEntityBelongedAddress": res.data.legalEntityBelongedAddress,
        "lebPeopleInCharge": res.data.lebPeopleInCharge,
        "lebPicPosition": res.data.lebPicPosition,
        "lebPicTelephone": res.data.lebPicTelephone,
        "lebSocialCreditCode": res.data.lebSocialCreditCode,
        "competentDepartmentName": res.data.competentDepartmentName,
        "competentDepartmentAddress": res.data.competentDepartmentAddress,
        "cdPeopleInCharge": res.data.cdPeopleInCharge,
        "cdPicPosition": res.data.cdPicPosition,
        "cdPicTelephone": res.data.cdPicTelephone,
        "characteristic": res.data.characteristic,
        "legalEntity": res.data.legalEntity
      })
    }, (err) => {
      //失败处理函数
      console.log(err)
    })
  },

  /**
   * 生命周期函数--监听页面隐藏
   */
  onHide: function () {

  },

  /**
   * 生命周期函数--监听页面卸载
   */
  onUnload: function () {

  },

  /**
   * 页面相关事件处理函数--监听用户下拉动作
   */
  onPullDownRefresh: function () {

  },

  /**
   * 页面上拉触底事件的处理函数
   */
  onReachBottom: function () {

  },

  /**
   * 用户点击右上角分享
   */
  onShareAppMessage: function () {

  }
})

File: server/cma/src/main/java/com/example/demo/Model/TestingInstitutionInformation.java
package com.example.demo.Model;
import javax.persistence.*;
@Entity
@Table(name = "TII_tables")
public class TestingInstitutionInformation {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    int id;
    private String testingInstitutionName;     //检测检验机构名称
    private String testingInstitutionAddress;       //地址
    private String postcode;       //邮编
    private String fax;        //传真
    private String email;        //Email地址
    private String tiPeopleInCharge;         //负责人
    private String tiPicPosition;        //负责人职务
    private String tiPicTelephone;         //负责人固定电话
    private String tiPicMobilephone;         //负责人手机
    private String liaison;         //联络人
    private String liaisonPosition;        //联络人职务
    private String liaisonTelephone;         //联络人固定电话
    private String liaisonMobilephone;         //联络人手机
    private String socialCreditCode;       //社会信用代码

    private String legalEntityBelongedName;     //所属法人单位名称
    private String legalEntityBelongedAddress;       //地址
    private String lebPeopleInCharge;         //负责人
    private String lebPicPosition;        //负责人职务
    private String lebPicTelephone;         //负责人电话
    private String lebSocialCreditCode;       //社会信用代码

    private String competentDepartmentName;     //主管部门名称
    private String competentDepartmentAddress;       //地址
    private String cdPeopleInCharge;         //负责人
    private String cdPicPosition;        //负责人职务
    private String cdPicTelephone;         //负责人电话

    private Byte characteristic;        //检测机构设施特点(0固定，1临时，2可移动，3多场所）
    private Byte legalEntity;        //法人类别，独立法人检验检测机构（0社团法人，1事业法人，2企业法人，3机关法人，4其他）检验检测机构所属法人（非独立法人检验检测机构填写此项）（5社团法人，6事业法人，7企业法人，8机关法人，9其他）

    public void setTiPicPosition(String tiPicPosition) {
        this.tiPicPosition = tiPicPosition;
    }

    public String getTiPicPosition() {
        return tiPicPosition;
    }

    public void setTiPeopleInCharge(String tiPeopleInCharge) {
        this.tiPeopleInCharge = tiPeopleInCharge;
    }

    public void setTestingInstitutionName(String testingInstitutionName) {
        this.testingInstitutionName = testingInstitutionName;
    }

    public void setTestingInstitutionAddress(String testingInstitutionAddress) {
        this.testingInstitutionAddress = testingInstitutionAddress;
    }

    public void setPostcode(String postcode) {
        this.postcode = postcode;
    }

    public void setFax(String fax) {
        this.fax = fax;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getTiPeopleInCharge() {
        return tiPeopleInCharge;
    }

    public String getTestingInstitutionName() {
        return testingInstitutionName;
    }

    public String getTestingInstitutionAddress() {
        return testingInstitutionAddress;
    }

    public String getPostcode() {
        return postcode;
    }

    public String getFax() {
        return fax;
    }

    public String getEmail() {
        return email;
    }

    public String getTiPicTelephone() {
        return tiPicTelephone;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public void setLebPicTelephone(String lebPicTelephone) {
        this.lebPicTelephone = lebPicTelephone;
    }

    public void setLebPicPosition(String lebPicPosition) {
        this.lebPicPosition = lebPicPosition;
    }

    public String getTiPicMobilephone() {
        return tiPicMobilephone;
    }

    public void setTiPicTelephone(String tiPicTelephone) {
        this.tiPicTelephone = tiPicTelephone;
    }

    public void setCdPicTelephone(String cdPicTelephone) {
        this.cdPicTelephone = cdPicTelephone;
    }

    public void setCdPicPosition(String cdPicPosition) {
        this.cdPicPosition = cdPicPosition;
    }

    public void setTiPicMobilephone(String tiPicMobilephone) {
        this.tiPicMobilephone = tiPicMobilephone;
    }

    public String getLebPicTelephone() {
        return lebPicTelephone;
    }

    public String getLebPicPosition() {
        return lebPicPosition;
    }

    public String getCdPicTelephone() {
        return cdPicTelephone;
    }

    public String getCdPicPosition() {
        return cdPicPosition;
    }

    public void setSocialCreditCode(String socialCreditCode) {
        this.socialCreditCode = socialCreditCode;
    }

    public void setLiaisonTelephone(String liaisonTelephone) {
        this.liaisonTelephone = liaisonTelephone;
    }

    public void setLiaisonPosition(String liaisonPosition) {
        this.liaisonPosition = liaisonPosition;
    }

    public void setLiaisonMobilephone(String liaisonMobilephone) {
        this.liaisonMobilephone = liaisonMobilephone;
    }

    public void setLiaison(String liaison) {
        this.liaison = liaison;
    }

    public void setLegalEntityBelongedName(String legalEntityBelongedName) {
        this.legalEntityBelongedName = legalEntityBelongedName;
    }

    public void setLegalEntityBelongedAddress(String legalEntityBelongedAddress) {
        this.legalEntityBelongedAddress = legalEntityBelongedAddress;
    }

    public void setLegalEntity(Byte legalEntity) {
        this.legalEntity = legalEntity;
    }

    public void setLebSocialCreditCode(String lebSocialCreditCode) {
        this.lebSocialCreditCode = lebSocialCreditCode;
    }

    public void setLebPeopleInCharge(String lebPeopleInCharge) {
        this.lebPeopleInCharge = lebPeopleInCharge;
    }

    public void setCompetentDepartmentName(String competentDepartmentName) {
        this.competentDepartmentName = competentDepartmentName;
    }

    public void setCompetentDepartmentAddress(String competentDepartmentAddress) {
        this.competentDepartmentAddress = competentDepartmentAddress;
    }

    public void setCharacteristic(Byte characteristic) {
        this.characteristic = characteristic;
    }

    public void setCdPeopleInCharge(String cdPeopleInCharge) {
        this.cdPeopleInCharge = cdPeopleInCharge;
    }

    public String getSocialCreditCode() {
        return socialCreditCode;
    }

    public String getLiaisonTelephone() {
        return liaisonTelephone;
    }

    public String getLiaisonPosition() {
        return liaisonPosition;
    }

    public String getLiaisonMobilephone() {
        return liaisonMobilephone;
    }

    public String getLiaison() {
        return liaison;
    }

    public String getLegalEntityBelongedName() {
        return legalEntityBelongedName;
    }

    public String getLegalEntityBelongedAddress() {
        return legalEntityBelongedAddress;
    }

    public String getLebSocialCreditCode() {
        return lebSocialCreditCode;
    }

    public String getLebPeopleInCharge() {
        return lebPeopleInCharge;
    }

    public String getCompetentDepartmentName() {
        return competentDepartmentName;
    }

    public String getCompetentDepartmentAddress() {
        return competentDepartmentAddress;
    }

    public String getCdPeopleInCharge() {
        return cdPeopleInCharge;
    }

    public Byte getLegalEntity() {
        return legalEntity;
    }

    public Byte getCharacteristic() {
        return characteristic;
    }
}



